server/stream.go

```diff
@@ -3,29 +3,32 @@ package server
 import (
 	"fmt"
 	"sync"
+	"time"
 
 	proto "github.com/liftbridge-io/liftbridge/server/protocol"
 )
 
 // stream is a message stream consisting of one or more partitions. Each
 // partition maps to a NATS subject and is the unit of replication.
 type stream struct {
-	name       string
-	subject    string
-	config     *proto.StreamConfig
-	partitions map[int32]*partition
-	resumeAll  bool // When partition(s) are paused, this indicates if all should be resumed
-	mu         sync.RWMutex
+	name         string
+	subject      string
+	config       *proto.StreamConfig
+	partitions   map[int32]*partition
+	resumeAll    bool // When partition(s) are paused, this indicates if all should be resumed
+	creationTime time.Time
+	mu           sync.RWMutex
 }
 
 // newStream creates a stream for the given NATS subject. All stream
 // interactions should only go through the exported functions.
 func newStream(name, subject string, config *proto.StreamConfig) *stream {
 	return &stream{
-		name:       name,
-		subject:    subject,
-		config:     config,
-		partitions: make(map[int32]*partition),
+		name:         name,
+		subject:      subject,
+		config:       config,
+		partitions:   make(map[int32]*partition),
+		creationTime: time.Now(),
```

Comment:
This might not do exactly what you want if you want the "real time" a stream was created. E.g. if the server is restarted, this timestamp will change to when the server recovers the stream. It will also be different for each server in the cluster.

I think if you wanted to ensure a consistent time that is persisted, you would need to have the leader set the creation time for the stream when it's replicated through Raft.

Reply:
Fixed in 8b1cb08.